spotify_player/src/cli/client.rs

```diff
@@ -344,9 +365,435 @@ async fn handle_playback_request(
     client.update_playback(state);
     Ok(())
 }
+
+async fn handle_playlist_request(
+    client: &Client,
+    state: &SharedState,
+    command: Command,
+) -> Result<String> {
+    match command {
+        Command::PlaylistNew {
+            name,
+            public,
+            collab,
+            description,
+        } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let id = user.id;
+
+            let resp = client
+                .spotify
+                .user_playlist_create(
+                    id,
+                    name.as_str(),
+                    Some(public),
+                    Some(collab),
+                    Some(description.as_str()),
+                )
+                .await?;
+            Ok(format!(
+                "Playlist '{}' with id '{}' was created.",
+                resp.name,
+                resp.id.to_string()
+            ))
+        }
+        Command::PlaylistDelete { id } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let uid = user.id;
+
+            let playlist_check = client
+                .spotify
+                .playlist_check_follow(id.to_owned(), &[uid])
+                .await;
+            if playlist_check.is_err() {
+                anyhow::bail!("Could not find playlist {}", id.to_string())
+            }
+
+            // Won't delete if not following
+            let following = playlist_check.unwrap().pop();
+            if following.unwrap() {
+                client.spotify.playlist_unfollow(id.to_owned()).await?;
+                Ok(format!("'{}' was deleted", id.to_string()))
+            } else {
+                Ok(format!("'{}' was not followed.", id.to_string()))
+            }
+        }
+        Command::PlaylistList => {
+            let resp = client
+                .spotify
+                .current_user_playlists_manual(Some(50), None)
+                .await?;
+
+            let mut out = String::new();
+            for pl in resp.items {
+                // Opposite way, don't know which is better
+                // Might want to add color
+                //out.push_str(format!("{}:{}\n", pl.name, pl.id.id()).as_str());
+                out.push_str(format!("{}:{}\n", pl.id.id(), pl.name).as_str());
+            }
+            out = out.trim().to_string();
+
+            Ok(out)
+        }
+        Command::PlaylistImport {
+            from: import_from,
+            to: import_to,
+        } => playlist_import(client, import_from, import_to).await,
+        Command::PlaylistFork { id } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let uid = user.id;
+
+            let from = client
+                .spotify
+                .playlist(
+                    id.to_owned(),
+                    None,
+                    Some(Market::Country(Country::UnitedStates)),
```

Comment:
Not everyone is from `US` should be `None` or at least `Market::FromToken` instead.